### ISSUE TITLE
Pruning some layers completely as result of layer importance

### ISSUE THREAD

**Issue by christian-steinmeyer** [zejiangh/Filter-GaP#6]
Did you ever come across the situation in which the layer ratios for a certain layer reaches 0 (or 1 - not sure which way around it is), so that the mask sets all channels to 0, essentially rendering the layer useless? How do you mitigate this?

Note, I reimplemented the paper in tensorflow, so I have slightly different code.

**Comment by vantienpham** [comment]
Hi, @christian-steinmeyer I'm also trying to reproduce the paper.
Would you like to collaborate?

**Comment by christian-steinmeyer** [comment]
What do you have in mind? Are you also working in Tensorflow?

**Comment by vantienpham** [comment]
@christian-steinmeyer At first I tried this repository but was unable to reproduce the reported result. I retrieve the original repository https://github.com/NVIDIA/DeepLearningExamples and have been working on this, which is coded in Pytorch and Nvidia's dependencies.

**Comment by christian-steinmeyer** [comment]
@pvtien96 I don't think we have too much in common to work with then.
